### PR TITLE
[build]: allow to customize USERNAME and PASSWORD in command line

### PR DESCRIPTION
**Pull request by lguohan**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,6 +39,8 @@ DOCKER_BUILD = docker build --no-cache \
 	    PLATFORM=$(PLATFORM) \
 	    BUILD_NUMBER=$(BUILD_NUMBER) \
 	    ENABLE_DHCP_GRAPH_SERVICE=$(ENABLE_DHCP_GRAPH_SERVICE) \
+	    PASSWORD=$(PASSWORD) \
```
Comment (marian-pritsak) [inline Makefile]: Need to check if they are not empty

**Comment by lguohan** [comment]
there are default values in the rules/config, am I missing anything here?

File: Makefile (hunk 1 above)
Comment (jleveque) [inline Makefile]: As Guohan mentioned below, there are default values in the rules/config, so the only way either value would be empty is if explicitly declared as such on the command line (e.g., `USERNAME=""`). This is a rare corner case and would only occur due to extreme user carelessness. I'm on the fence as to whether we need to check for it.

**Comment by jleveque** [comment]
Build failures: `Error: no or empty USERNAME argument`

File: Makefile (hunk 1 above)
Comment (jleveque) [inline Makefile]: See below. It appears that Make is already checking for missing/empty values.

Comment (taoyl-ms) [inline Makefile]: I think you misunderstood marian's comment. What is happening here is that adding this statement will override slave.mk's $PASSWORD with Makefile's $PASSWORD, even if it is empty. So we DO need to check empty before passing it or the value in rule.cofig will be overridden.

What is different here from ENABLE_DHCP_GRAPH_SERVICE is that in the case of ENABLE_DHCP_GRAPH_SERVICE actually empty is a valid value (and also the default value), while for $PASSWORD it is not.

Comment (jleveque) [inline Makefile]: Ah, gotcha. Pretty sure Guohan's intention was to have them initially set to the defaults in rules/config and only overwritten if explicitly specified on the command line.

Comment (lguohan) [inline Makefile]: resolved

**Comment by lguohan** [comment]
@marian-pritsak, please review

File: rules/config [hunk 2]
```diff
@@ -27,10 +27,10 @@ SONIC_CONFIG_LOG_TO_FILES = y
 # SONIC_CONFIG_ENABLE_COLORS = y
 
 # USERNAME - username for installer build
```
Comment (jleveque) [inline rules/config]: Should change variable name in comments to `DEFAULT_USERNAME` to match.

File: rules/config [hunk 3]
```diff
@@ -27,10 +27,10 @@ SONIC_CONFIG_LOG_TO_FILES = y
...
-USERNAME = admin
+DEFAULT_USERNAME = admin
 
 # PASSWORD - password for installer build
```
Comment (jleveque) [inline rules/config]: Should change variable name in comments to `DEFAULT_PASSWORD` to match.

**Pull request merged**